src/app/beer_garden/systems.py

```diff
@@ -147,34 +149,36 @@ def reload_system(system_id: str) -> None:
 
     Args:
         system_id: The System ID
+        system: The System
 
     Returns:
         None
     """
     # TODO - It'd be nice to have a check here to make sure system is managed
 
-    return db.query_unique(System, id=system_id)
+    return system or db.query_unique(System, id=system_id)
 
 
 @publish_event(Events.SYSTEM_REMOVED)
-def remove_system(system_id: str) -> System:
+def remove_system(system_id: str = None, system: System = None) -> System:
     """Remove a system
 
     Args:
         system_id: The System ID
+        system: The System
 
     Returns:
         The removed System
 
     """
-    system = db.query_unique(System, id=system_id)
+    system = system or db.query_unique(System, id=system_id)
 
     db.delete(system)
```

Comment:
Had an error when attempting to delete a System. I don't know if is this caused by the System object being passed in or something else.

```
mongoengine.errors.ValidationError: 'default:echo-1.0.0.dev0' is not a valid ObjectId, it must be a 12-byte input or a 24-character hex string
```

Reply:
Whoops. Needed to be `db.delete(system=system)`